File: src/main/java/app/askresume/api/member/facade/MyMemberFacade.java
package app.askresume.api.member.facade;

import app.askresume.domain.member.entity.Member;
import app.askresume.domain.member.service.MemberService;
import app.askresume.api.member.dto.request.ModifyInfoRequest;
import app.askresume.api.member.dto.response.MemberInfoResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
@Transactional(readOnly = true)
public class MyMemberFacade {

    private final MemberService memberService;


    public MemberInfoResponse findMemberInfo(Long memberId) {
        Member member = memberService.findMemberById(memberId);
        return MemberInfoResponse.of(member);
    }

    @Transactional
    public void modifyMemberInfo(Long memberId, ModifyInfoRequest request) {
        Member member = memberService.findMemberById(memberId);
        memberService.modifyMemberInfo(member, request);
    }

    @Transactional
    public void secessionMember(Long memberId) {
        Member member = memberService.findMemberById(memberId);
        memberService.secessionMember(member);
    }
}

File: src/main/kotlin/app/askresume/api/member/facade/MyMemberFacade.kt
package app.askresume.api.member.facade

import app.askresume.api.member.mapper.MyMemberMapper
import app.askresume.api.member.vo.MemberInfoResponse
import app.askresume.api.member.vo.ModifyInfoRequest
import app.askresume.domain.member.service.MemberCommandService
import app.askresume.domain.member.service.MemberReadOnlyService
import app.askresume.global.annotation.Facade
import org.springframework.transaction.annotation.Transactional

@Facade
@Transactional(readOnly = true)
class MyMemberFacade(
    private val memberReadOnlyService: MemberReadOnlyService,
    private val memberCommandService: MemberCommandService,

    private val myMemberMapper: MyMemberMapper,
) {

    fun findMyMemberInfo(memberId: Long): MemberInfoResponse {
        val memberInfoDto = memberReadOnlyService.findMemberInfo(memberId) !!
        return myMemberMapper.memberInfoDtoToMemberInfoResponse(memberInfoDto)
    }

    @Transactional
    fun modifyMyMemberInfo(memberId: Long, request: ModifyInfoRequest) {
        val memberInfoDto = memberReadOnlyService.findMemberInfo(memberId) !!
        memberCommandService.modifyMemberInfo(
            memberId = memberInfoDto.memberId,
            username = request.username,
            profile = request.profile,
        )
    }

    @Transactional
    fun secessionMember(memberId: Long) {
        memberCommandService.secessionMember(memberId)
    }
}

